controllers/asosecret_controller.go

```diff
@@ -278,14 +278,19 @@ func (asos *ASOSecretReconciler) createSecretFromClusterIdentity(ctx context.Con
 	newASOSecret.Data["AZURE_TENANT_ID"] = []byte(identity.Spec.TenantID)
 	newASOSecret.Data["AZURE_CLIENT_ID"] = []byte(identity.Spec.ClientID)
 
+	// If the identity type is WorkloadIdentity, then we don't need to fetch the secret so return early
+	if identity.Spec.Type == infrav1.WorkloadIdentity {
```

Comment:
what about `UserAssignedMSI` identity type? wouldn't that also not have a secret?

Reply:
I was wondering if it might be worth making that change alongside the other necessary changes for UserAssignedMSI like creating the AzureIdentity resources as part of #3743.